.github/workflows/jobs-poster.yaml

```diff
@@ -0,0 +1,78 @@
+name: Post New Jobs
+
+on:
+  pull_request: []
+  push:
+    paths:
+      - '_data/jobs.yml'
+    branches:
+      - main
+
+jobs:
+  slack-poster:
+    runs-on: ubuntu-latest
+    name: Run Jobs Slack Poster
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+
+      - id: updater
+        name: Job Updater
+        uses: rseng/jobs-updater@add/deploy-arg
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
+        with:
+          filename: "_data/jobs.yml"
+          key: "url"
+          deploy: false
+          test: true
+
+      - run: echo ${{ steps.updater.outputs.fields }}
+        name: Show New Jobs
+        shell: bash
+
+      - run: echo ${{ steps.updater.outputs.matrix }}
+        name: Show Matrix
+        shell: bash
+
+    outputs:
+      matrix: ${{ steps.updater.outputs.matrix }}
+      fields: ${{ steps.updater.outputs.fields }}
+        
+  twitter-poster:
+    needs:
+      - slack-poster
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix: 
+        jobs: ${{ fromJson(needs.slack-poster.outputs.matrix) }}
+
+    name: Prepare to Post to Twitter
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v2
+
+    - name: Generate Message
+      run: |
+         emoji=$(python -c "import random;print(random.choice(['⭐️','😍️','❤️','👀️','✨️','🤖️','💼️','🤩️','😸️','😻️','👉️','🕶️','🔥️','💻️']))")
+         message="New Job! ${emoji} ${{ matrix.jobs[0] }}"
+         echo "message=${message}" >> $GITHUB_ENV         
+
+    - name: View Jobs Message
+      env:
+        message: ${{ env.message }}
+      run: |
+         echo "Job message for Twitter:"
+         echo ${message}
+
+#    - uses: Eomm/why-don-t-you-tweet@bfa8227517f7d2351aa9f2bc48b312792f2ccbf7
```

Comment:
This is some kind of software dependency? I wonder if it might be worth creating an issue to keep track somewhere of what all we depend on where.

Reply:
It's a GitHub action, and the way GitHub actions work is that they are used directly from that GitHub repo. I pin at a particular commit because I reviewed that commit for security.